### PR TITLE
Update Client wrapper 

### PR DESCRIPTION
**Pull request by odinuv**
https://keboola.atlassian.net/browse/PS-3098
Refactored tests because the client wrapper is now immutable and branch cannot be set after construction.

Depends on 
- https://github.com/keboola/storage-api-php-client-branch-wrapper/pull/9
- https://github.com/keboola/input-mapping/pull/117

### CODE REVIEW

File: tests/InitSynapseStorageClientTrait.php [hunk 1]
```diff
@@ -32,18 +32,21 @@ protected function checkSynapseTests()
...
+                (string) getenv('SYNAPSE_STORAGE_API_TOKEN'),
+                null,
+                null,
+                null,
+                null,
+                1,
+                null,
+                null,
+                function () {
                     return 1;
```
Comment (MiroCillik) [inline tests/InitSynapseStorageClientTrait.php:46]: Skoda funkce na tohle :D

File: tests/InitSynapseStorageClientTrait.php [hunk 2]
```diff
@@ -32,18 +32,21 @@ protected function checkSynapseTests()
...
-                },
-            ]),
-            null,
-            null
+                }
+            )
         );
-        $clientWrapper->setBranchId('');
```
Comment (MiroCillik) [inline tests/InitSynapseStorageClientTrait.php:46]: 🎊

File: tests/Writer/BaseWriterTest.php [hunk 3]
```diff
@@ -63,21 +56,24 @@ public function setUp()
...
+                STORAGE_API_TOKEN,
+                $branchId,
+                null,
+                null,
+                null,
+                1,
+                null,
+                null,
+                function () {
                     return 1;
```
Comment (ErikZigo) [inline tests/Writer/BaseWriterTest.php:73]: Jen poznamka. Tohle je peklo a vubec by to nemelo mit volitelny paremetry v konstruktoru kdyz jich je takhle bambilion.

**Review by ErikZigo — APPROVED** [review]

**Review by MiroCillik — APPROVED** [review]
👍

File: tests/Writer/BaseWriterTest.php (hunk 3 above)
Comment (odinuv) [reply tests/Writer/BaseWriterTest.php:73]: No to je škaredý zápis, ale můžu to zapsat takhle:
```
        $clientOptions = (new ClientOptions())
            ->setUrl(STORAGE_API_URL)
            ->setToken(STORAGE_API_TOKEN)
            ->setBranchId($branchId)
            ->setBackoffMaxTries(1)
            ->setJobPollRetryDelay(function () {
                return 1;
            });
        $this->clientWrapper = new ClientWrapper($clientOptions);
```
lepší?

Comment (ErikZigo) [reply tests/Writer/BaseWriterTest.php]: jo lepsi

**Pull request merged**